zerver/lib/upload/s3.py

```diff
@@ -138,6 +139,21 @@ def get_boto_client(self) -> S3Client:
             )
         return self._boto_client
 
+    def move_file_inside_bucket(
+        self, old_key: str, new_key: str, bucket: Optional[Bucket] = None
+    ) -> bool:
+        if bucket is None:
+            bucket = self.uploads_bucket
+        copy_source: CopySourceTypeDef = {"Bucket": bucket.name, "Key": old_key}
+        try:
+            bucket.copy(copy_source, new_key)
+            self.delete_file_from_s3(old_key, self.uploads_bucket)
```

Comment:
In your commit message you say "there is no optimal method" but this is pretty much the canonical form.  Internally, they have implemented their "copy" as something which is effectively O(1) time, regardless of filesize, so both of these operations should be fast.

Reply:
My bad, Ig by optimal I meant direct method to do so, and I didn't know that copy takes O(1) time. I will update the commit message.

Reply:
Hm -- I though I recalled copies were O(1) within a bucket, but I actually can't find a citation for that.  But it doesn't do any data transfer through the client, at least, so it should be many GB/sec regardless of the network connection between the Zulip server and S3 -- it's all within S3's network.

Reply:
Updated the commit message.